nemo/collections/asr/jasper.py

```diff
@@ -317,3 +317,179 @@ def forward(self, encoder_output):
+        init_mode (str): Describes how neural network parameters are
+            initialized. Options are ['xavier_uniform', 'xavier_normal',
+            'kaiming_uniform','kaiming_normal'].
+            Defaults to "xavier_uniform".
+    """
+
+    @property
+    def input_ports(self):
+        """Returns definitions of module input ports.
+
+        encoder_output:
+            0: AxisType(BatchTag)
+
+            1: AxisType(EncodedRepresentationTag)
+
+            2: AxisType(ProcessedTimeTag)
+        """
+
+        return {"encoder_output": NeuralType(('B', 'D', 'T'), AcousticEncodedRepresentation())}
+
+    @property
+    def output_ports(self):
+        """Returns definitions of module output ports.
+
+        logits:
+            0: AxisType(BatchTag)
+
+            1: AxisType(ChannelTag)
+                
+        embs: 
+            0: AxisType(BatchTag)
+            1: AxisType(EncodedRepresentationTah) 
+        """
+        return {
+            "logits": NeuralType(('B', 'D'), LogitsType()),
+            "embs": NeuralType(('B', 'D'), AcousticEncodedRepresentation()),
+        }
+
+    def __init__(
+        self, feat_in, num_classes, emb_sizes=[1024, 1024], gram=False, super_vector=True, init_mode="xavier_uniform"
+    ):
+        TrainableNM.__init__(self)
+        self._feat_in = 0
+        if gram:
+            self._feat_in += feat_in ** 2
+        else:
+            self._feat_in += 2 * feat_in
+
+        if super_vector and gram:
+            self._feat_in += 2 * feat_in
+
+        self._midEmbd1 = int(emb_sizes[0])  # Spkr Vector Embedding Shape
+        self._midEmbd2 = int(emb_sizes[1]) if len(emb_sizes) > 1 else 0  # Spkr Vector Embedding Shape
+
+        self._num_classes = num_classes
+        self._pooling = StatsPoolLayer(gram=gram, super_vector=super_vector)
+        self.norm = nn.BatchNorm1d(feat_in)
+        self.mid1 = self.affineLayer(self._feat_in, self._midEmbd1, learn_mean=False)
+        self.mid2 = self.affineLayer(self._midEmbd1, self._midEmbd2, learn_mean=False)
+        self.final = nn.Linear(self._midEmbd2, self._num_classes)
+
+        self.apply(lambda x: init_weights(x, mode=init_mode))
+        self.to(self._device)
+
+    def affineLayer(self, inp_shape, out_shape, learn_mean=True):
+        layer = nn.Sequential(
+            nn.Linear(inp_shape, out_shape),
+            nn.BatchNorm1d(out_shape, affine=learn_mean, track_running_stats=True),
+            nn.ReLU(),
+        )
+
+        return layer  # layer, embs
+
+    def forward(self, encoder_output):
+        # encoder_output = self.norm(encoder_output)
+        pool = self._pooling(encoder_output)
+        mid1, emb1 = self.mid1(pool), self.mid1[:2](pool)
+        mid2, embs = self.mid2(mid1), self.mid2[:2](mid1)
+        out = self.final(mid2)
+
+        return out, emb1
+
+
+# Siamese Network, support to be added in future releases
+class SiameseDecoderForSpeakerClass(TrainableNM):
```

Comment:
Does this class work? If not, can you comment it out for now?

Reply:
okay